File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    
    n_count = str(bin(n)[2:]).count("1")
    i = 1
    while True:
        next_num = n+i
        if str(bin(next_num)[2:]).count("1") == n_count:
            return next_num
        else:
            i+=1

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        int next = n+1;
        while(true){
            String b1 = Integer.toBinaryString(n);
            String b2 = Integer.toBinaryString(next);
            if(count(b1) == count(b2)){
                break;
            }
            next++;
        }
        
        return next;
    }
    
    public int count(String bi){
        int count = 0;
        for(int i = 0;i<bi.length();i++){
            if(bi.charAt(i) == '1'){
                count++;
            }
        }
        return count;
    }
}